.gitignore

```diff
@@ -167,3 +167,9 @@ tags
 
 # ruff
 .ruff_cache
+
+# folders generated in test_trainer.py
+/test
+/regression
+/None
+/examples/checkpoint-*
```

Comment:
Do you know which test is causing this. ? We try to put everything that are saved in temp folder

Reply:
Thank you for your comment! Let me see...

I've searched the lines maybe related to:

### ".test"
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1193
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1454
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1476
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1508
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1516
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1579
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1623
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1682

### "./regression":
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L555
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L738
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L990
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L1009


### "./examples"
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L3188
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L3206
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L3240
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L3255
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L3282


### "./None"
I couldn't find any directly specified part, but I've seen "./None" folder is made every time I run `tests_trainer`

Reply:
I'm not sure all of these are the main cause, but when I ran `test_trainer.py`, it surely made the folders I wrote in .gitignore.

Reply:
I found 117 lines which use `with tempfile.TemporaryDirectory() as tmp_dir:` and pass tmp_dir to TrainingArguments, so most of the test experiments properly utilizes temp dir, and the lines I picked above are irregular.

Reply:
I found another non-tmp-dir specified in `test_trainer.py`

### "./generation"
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L786
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L801
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L883
https://github.com/huggingface/transformers/blob/bc6ae0d55e11e46eaed4da71b6bc5087d38cec70/tests/trainer/test_trainer.py#L898